### DISCUSSION TITLE
using arcgisbinding::arc.open() with arcgislayers

### DISCUSSION THREAD

**Discussion by Nova-Scotia** [R-ArcGIS/arcgislayers#79] [Q&A] [answered]
Hi @JosiahParry ,

Unfortunately I don't have a reproducible example but I thought you might have seen this error in testing enough that you might be able to tell me where I'm tripping up.

I've published a hosted feature layer in AGOL and when I try to add points to it with {arcgislayers} - which was working yesterday - it's giving me an error message. I basically copied and pasted code that I had working yesterday, so I'm puzzled. I deleted the old feature layers I was working with, so can't compare behaviour or see if my old code is still working. Hoping this is something really obvious I'm missing... 

```
# grab animal locations data from AGOL - they are in WGS84/3857
agolpts_sp_ref <- arc.open("https://services1.arcgis.com/blahblahblah/arcgis/rest/services/Animal_locations/FeatureServer/0")

# make my data spatial
newpts_sp <- st_as_sf(newpts, 
                          coords = c("location_long", "location_lat"), 
                          crs = wgs84)

# transform to 3857 just to be certain 
newpts_sp <- st_transform(newpts_sp, 3857)

# Get token
auth_binding() |>
     set_auth_token() 

# add these to my features in AGOL
result <- add_features(agolpts_sp_ref, newpts_sp, match_on = "name")

Error in as.character.default(new("arc.feature_impl", shapeinfo = list( : 
  no method for coercing this S4 class to a vector
```

**Comment by JosiahParry** [comment, marked as answer]
@Nova-Scotia it looks like youre mixing the `arcgisbinding::arc.open()` with the `arcgislayers::arc_open()`. To use {arcgislayers} functions such as `add_features()` you will need to use `arc_open()` here. Let me know if that fixes the issue! 

SImilar syntax but different :P

**Reply by Nova-Scotia** [reply]
Yeah, I'm going to owe you lunch by the time I'm finished writing this code. Thanks! That worked.

**Reply by JosiahParry** [reply]
Woooohhh!!
